### PR TITLE
Supported regex and glob in filter processor

### PR DESCRIPTION
**Pull request by BenMatase**
Added support for regex and glob matchers

### CODE REVIEW

File: plugins/processors/t128_filter/README.md [hunk 1]
```diff
@@ -9,12 +9,23 @@ The `t128_filter` filters out metrics passing through it. This processor is usef
...
+    ##  * "exact": exact string comparison
+    ##  * "glob": go flavored glob comparison (see https://github.com/gobwas/glob)
+    ##  * "regex": go flavored regex comparison
+    # mode = "exact"
 
   [processors.t128_filter.condition.tags]
-     #tag1 = ["value1", "value2"]
-     #tag2 = ["value3"]
+    # tag1 = ["value1", "value2"]
+	  # tag2 = ["value3"]
```
Comment (gregschrock) [inline plugins/processors/t128_filter/README.md]: is this indent intentional?

File: plugins/processors/t128_filter/t128_filter.go [hunk 2]
```diff
@@ -10,20 +14,36 @@ const sampleConfig = `
...
+	##  * "exact": exact string comparison
+	##  * "glob": go flavored glob comparison (see https://github.com/gobwas/glob)
+	##  * "regex": go flavored regex comparison
+    # mode = "exact"
 
   [processors.t128_filter.condition.tags]
-     #tag1 = ["value1", "value2"]
-	 #tag2 = ["value3"]
+     # tag1 = ["value1", "value2"]
+	 # tag2 = ["value3"]
```
Comment (gregschrock) [inline plugins/processors/t128_filter/t128_filter.go]: intentional indent?

File: plugins/processors/t128_filter/t128_filter.go [hunk 3]
```diff
@@ -10,20 +14,36 @@ const sampleConfig = `
...
   [processors.t128_filter.condition.tags]
-     #tag1 = ["value3"]
+     # tag1 = ["value3"]
 `
 
 type tags map[string][]string
+type mode string
+
+const (
+	EMPTY_MODE mode = ""
```
Comment (gregschrock) [inline plugins/processors/t128_filter/t128_filter.go]: 1. These don't follow the typical go naming convention (camel case)
2. It's totally personal preference, but I like to use `modeEmpty`, `modeExact` etc. so autocomplete will show me all the available modes. Take it or leave it.

File: plugins/processors/t128_filter/t128_filter.go [hunk 4]
```diff
@@ -66,6 +86,46 @@ func (m exactMatcher) Matches(point telegraf.Metric) bool {
 	return false
 }
 
+type regexMatcher struct {
+	tag         string
+	expressions []*regexp.Regexp
+}
+
+func (m regexMatcher) Matches(point telegraf.Metric) bool {
+	value, ok := point.GetTag(m.tag)
```
Comment (gregschrock) [inline plugins/processors/t128_filter/t128_filter.go:95]: Is there a way to say a tag may be one of a few values or missing? Just curious if that's a valid use case we should think through.

File: plugins/processors/t128_filter/t128_filter.go [hunk 5]
```diff
@@ -97,24 +157,82 @@ func (c orConjMatcher) Matches(point telegraf.Metric) bool {
...
 	}
 
 	return orConjMatcher{conditionMatchers}, nil
 }
 
-func getTagMatchers(tags tags) []matcher {
+func getTagMatchers(tags tags, mode mode) ([]matcher, error) {
 	tagMatchers := make([]matcher, len(tags))
 
 	j := 0
```
Comment (gregschrock) [inline plugins/processors/t128_filter/t128_filter.go]: I'm just noticing this now. I think avoiding this index would be less error prone.
```go
tagMatchers := make([]matcher, 0, len(tags))
...
tagMatchers = append(tagMatchers, matcher)
```

File: plugins/processors/t128_filter/t128_filter.go [hunk 6]
```diff
@@ -97,24 +157,82 @@ func (c orConjMatcher) Matches(point telegraf.Metric) bool {
...
+	case GLOB_MODE:
+		globs, err := compileGlobs(values)
+		if err != nil {
+			return nil, err
+		}
+
+		return globMatcher{tag, globs}, nil
+	}
+
+	fmt.Println("hi")
```
Comment (gregschrock) [inline plugins/processors/t128_filter/t128_filter.go]: hey

**Review by gregschrock — APPROVED** [review]
This is SWEET!!

File: plugins/processors/t128_filter/t128_filter.go (hunk 4 above)
Comment (BenMatase) [reply plugins/processors/t128_filter/t128_filter.go:95]: Yea, I remember you mentioning that before. I'll create a jira to add an option for that https://128technology.atlassian.net/browse/MON-377

File: plugins/processors/t128_filter/t128_filter.go (hunk 3 above)
Comment (BenMatase) [reply plugins/processors/t128_filter/t128_filter.go]: Oh true, I was getting used to Mist's way of doing the enums in their oc term repo, I'll change

File: plugins/processors/t128_filter/t128_filter.go (hunk 2 above)
Comment (BenMatase) [reply plugins/processors/t128_filter/t128_filter.go]: Weird, these indents looked fine in vscode before I pushed... I'll fix

Comment (gregschrock) [reply plugins/processors/t128_filter/t128_filter.go]: probably tabs vs. spaces

**Review by gregschrock — APPROVED** [review]

**Review by salegria8 — APPROVED** [review]
good stuff!

File: plugins/processors/t128_filter/README.md [hunk 7]
```diff
@@ -9,12 +9,23 @@ The `t128_filter` filters out metrics passing through it. This processor is usef
 ```toml
 [[processors.t128_filter]]
   ## The conditions that must be met to pass a metric through. This is similar
-  ## behavior to a tagpass, but multiple tags are ANDed
+  ## behavior to a tagpass, but the multiple tags are ANDed
   [[processors.t128_filter.condition]]
+    ## Mode dictates how to match the condition's tag values
+    ## Valid values are:
+    ##  * "exact": exact string comparison
+    ##  * "glob": go flavored glob comparison (see https://github.com/gobwas/glob)
```
Comment (agrawalkaushik) [inline plugins/processors/t128_filter/README.md:17]: I know this links to the github page but I think having some examples of globs and regex would be useful.

**Pull request merged**